MinimalCirclesPathTest.py

```diff
@@ -0,0 +1,61 @@
+from MinimalCirclesPath import *
+import unittest
+
+class TestMinimalCirclesPath(unittest.TestCase):
+
+	# tests gets distance betweeen two points (haversine)
+	# test data from https://www.nhc.noaa.gov/gccalc.shtml
+	def testGetDistance(self):
+		
+		# DFW to DEN
+		self.assertAlmostEqual(getDistance(32.897, -97.038, 39.862, -104.673), 557, places = 0)
+
+		# TUL to WSG
+		self.assertAlmostEqual(getDistance(36.198, -95.888, 40.167, -80.25), 774, places = 0)
+
+		# RSW to LAX 
+		self.assertAlmostEqual(getDistance(26.542, -81.755, 33.943, -118.407), 1940, places = -1)
+
+		# HAX to MKO (two airports close in Muskogee Oklahoma)
+		self.assertAlmostEqual(getDistance(35.746, -95.413, 35.667, -95.367), 5, places = 0)
+
+		# japan to germany (CTS to ZVM)
+		self.assertAlmostEqual(getDistance(42.8, 141.667, 49.467, -96.833), 4470, places = -1)
+
+
+	# tests method that gets bearing given two points
+	# test data from https://www.movable-type.co.uk/scripts/latlong.html
+	def testCalculateBearing(self):
+		
+		# DFW to DEN
+		self.assertAlmostEqual(calculateBearing(32.897, -97.038, 39.862, -104.673), 320.78, places = 2)
+
+
+	# tests that the final NOTAM covers the furthest point we want covered
+	def testGetPathFinalPoint(self):
+
+		# DFW to DEN
+		radius = 100
+		destLat=39.862
+		destLong=-104.673
+		pathWidth=50
+		bearing = 320.7808955450159
+
+		# gets the final point we want covered
+		updateDest = nextPoint(destLat, destLong, bearing, (pathWidth/2))
+
+		# gets all the points for NOTAMs we would cover from DFW to DEN
+		DFW_DEN = getPath(startLat=32.897, startLong=-97.038, destLat=destLat, destLong=destLong, radius=radius, pathWidth=pathWidth)
+
+		# checks if the final NOTAM includes the last point we want covered
+		self.assertGreater(radius, getDistance(DFW_DEN[-1][0], DFW_DEN[-1][1], updateDest[0], updateDest[1]))
```

Comment:
I think this is a very good and simple check that our path calculation ends up in the right spot.  However, I also think there is an insidious bug hiding in your code that's gone undetected (which is part of the reason why I've been pushing for some unit tests).  You've happened to pick a flight path (from Denver to Dallas) for which I think this check here passes, but I'd like to suggest you test other paths (not just this one).  For example, paths that are mostly eastward/westward vs. paths which are mostly northward/southward, like a path from Seattle (KSEA) to Duluth (KDLH):
![image](https://github.com/Gabrieldowen/NOTAMyWatch/assets/19157011/a3f7eb21-10a6-456c-a247-0db58bc9e59b)

Reply:
by the way, that image came from skyvector.com which is another helpful pilot resource that shows airport information and helps with flight path planning (but it doesn't do anything with NOTAMs)

Reply:
I believe I fixed the bug in my latest commit. It involves updating the bearing at each point you move towards the destination

Reply:
that's certainly one way to solve the bug, arguably the most straight-forward way, though it may warrant a discussion about what the impact of that implementation decision is and whether alternatives exist which may be better